src/components/NumberFormat/index.tsx

```diff
@@ -0,0 +1,13 @@
+import {observer} from 'mobx-react';
+import * as numeral from 'numeral';
+import * as React from 'react';
+
+interface NumberFormatProps {
+  value: number;
+}
+
+export const NumberFormat: React.SFC<NumberFormatProps> = ({value}) => (
```

Comment:
good idea, but here are my thoughts:
1) you don't need `observer` here cause you are already dereferenced the value, check https://github.com/mobxjs/mobx-react#customizing-inject and https://github.com/mobxjs/mobx/blob/gh-pages/docs/best/react-performance.md#dereference-values-late. It's even better to not depend on `mobx`
2) **format** should be provided as _prop_ with default format
3) it would be better to call it just `Number`

Reply:
Number variable clashes with existing number type, so I left it as NumberFormat. Other improvements have been introduced.